component/src/main/java/org/wso2/extension/siddhi/map/avro/sourcemapper/AvroSourceMapper.java

```diff
@@ -0,0 +1,424 @@
+                            case FLOAT:
+                                if (JsonToken.VALUE_NUMBER_FLOAT.equals(jsonToken) ||
+                                        JsonToken.VALUE_NUMBER_INT.equals(jsonToken)) {
+                                    data[position] = attributeConverter.getPropertyValue(parser.getValueAsString(),
+                                            Attribute.Type.FLOAT);
+                                } else {
+                                    log.error("Avro message " + avroMessage +
+                                            " contains incompatible attribute types and values. Value " +
+                                            parser.getText() + " is not compatible with type FLOAT. " +
+                                            "Hence dropping the message.");
+                                    return new Event[0];
+                                }
+                                break;
+                            case LONG:
+                                if (JsonToken.VALUE_NUMBER_INT.equals(jsonToken)) {
+                                    data[position] = parser.getValueAsLong();
+                                } else {
+                                    log.error("Json message " + avroMessage +
+                                            " contains incompatible attribute types and values. Value " +
+                                            parser.getText() + " is not compatible with type LONG. " +
+                                            "Hence dropping the message.");
+                                    return new Event[0];
+                                }
+                                break;
+                            case OBJECT:
+                                switch (jsonToken) {
+                                    case START_OBJECT:
+                                    case START_ARRAY:
+                                        jsonObjectNode = objectMapper.readTree(avroMessage).findValue(key);
+                                        data[position] = gson.fromJson(jsonObjectNode.toString(), Object.class);
+                                        handleJsonObject(jsonObjectNode, parser);
+                                        break;
+                                    case VALUE_STRING:
+                                        data[position] = parser.getValueAsString();
+                                        break;
+                                    case VALUE_NUMBER_INT:
+                                        data[position] = parser.getValueAsInt();
+                                        break;
+                                    case VALUE_NUMBER_FLOAT:
+                                        data[position] = attributeConverter.getPropertyValue(parser.getValueAsString(),
+                                                Attribute.Type.FLOAT);
+                                        break;
+                                    case VALUE_TRUE:
```

Comment:
This also because when the case is either TRUE or FALSE the same line of code need to be implemented. That is data[position] = parser.getValueAsBoolean();. When case TRUE it will implemenet whatever in case FALSE, and case FALSE has a break.